File: adt2keyb/sdl/TranslateKeycodes.c
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2024 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

void TranslateKeycodes (void) {
  SDLKey k;
  SDLMod m;

  // Translate SDL_Keycodes to scancodes
  for (k = 0; k <= SDLK_LAST; k++) {
    unsigned i;

    for (i = 0; i < SYMTABSIZE; i++)
      if (symtab[i].SDLK == k)
        keydown[symtab[i].scan] = keystate[k];
  }

  // Read `Caps Lock' and `Num Lock' state
  m = SDL_GetModState ();
  _capslock = (m & KMOD_CAPS) != 0;
  _numlock = (m & KMOD_NUM) != 0;
}

File: adt2keyb/sdl/pas/TranslateKeycodes.pas
// This file is part of Adlib Tracker II (AT2).
//
// SPDX-FileType: SOURCE
// SPDX-FileCopyrightText: 2014-2024 The Adlib Tracker 2 Authors
// SPDX-License-Identifier: GPL-3.0-or-later

procedure TranslateKeycodes;

var
  i,j: Integer;
  modkeys: SDLMod;

begin
  // translate SDL_Keycodes to scancodes
  For i := 0 to SDLK_LAST do
    For j := 0 to PRED(SYMTABSIZE) do
      If (i = symtab[j*10+9]) then
        keydown[symtab[j*10]] := keystate^[i];

  // read capslock and numlock state
  modkeys := SDL_GetModState;
  _capslock := (modkeys AND KMOD_CAPS) <> 0;
  _numlock := (modkeys AND KMOD_NUM) <> 0;
end;
